source/patterns/@aws-solutions-constructs/aws-events-rule-sqs/lib/index.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ *  Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as sqs from '@aws-cdk/aws-sqs';
+import * as events from '@aws-cdk/aws-events';
+import * as defaults from '@aws-solutions-constructs/core';
+import { ArnPrincipal } from '@aws-cdk/aws-iam';
+import { Construct } from '@aws-cdk/core';
+import { overrideProps } from '@aws-solutions-constructs/core';
+
+/**
+ * @summary The properties for the EventsRuleToSQS Construct
+ */
+export interface EventsRuleToSQSProps {
+  /**
+   * Existing instance of SQS queue object, if this is set then the queueProps is ignored.
+   *
+   * @default - None
+   */
+  readonly existingQueueObj?: sqs.Queue,
+  /**
+   * User provided props to override the default props for the SQS queue.
+   *
+   * @default - Default props are used
+   */
+  readonly queueProps?: sqs.QueueProps,
+  /**
+   * Optional user provided properties for the dead letter queue
+   *
+   * @default - Default props are used
+   */
+  readonly deadLetterQueueProps?: sqs.QueueProps,
+  /**
+   * Whether to deploy a secondary queue to be used as a dead letter queue.
+   *
+   * @default - true.
+   */
+  readonly deployDeadLetterQueue?: boolean,
+  /**
+   * The number of times a message can be unsuccessfully dequeued before being moved to the dead-letter queue.
+   *
+   * @default - required field if deployDeadLetterQueue=true.
+   */
+  readonly maxReceiveCount?: number
+  /**
+   * User provided eventRuleProps to override the defaults
+   *
+   * @default - None
+   */
+  readonly eventRuleProps: events.RuleProps
+}
+
+export class EventsRuleToSQS extends Construct {
+  public readonly sqsQueue: sqs.Queue;
+  public readonly deadLetterQueue: sqs.DeadLetterQueue | undefined;
+  public readonly eventsRule: events.Rule;
+  // public readonly key: kms.Key
+  /**
+   * @summary Constructs a new instance of the EventsRuleToSQS class.
+   * @param {cdk.App} scope - represents the scope for all the resources.
+   * @param {string} id - this is a a scope-unique id.
+   * @param {EventsRuleToSQSProps} props - user provided props for the construct
+   * @since 0.8.0
+   * @access public
+   */
+  constructor(scope: Construct, id: string, props: EventsRuleToSQSProps) {
+    super(scope, id);
+
+    // Setup the dead letter queue, if applicable
+    if (props.deployDeadLetterQueue || props.deployDeadLetterQueue === undefined) {
```

Comment:
Check if the user has provided `existingQueueObj?` then skip, also add an integ test case for this scenario if user provides an `existingQueueObj?` and no value for `deployDeadLetterQueue?` it should not deploy the DLQ.. this might be an issue with the other existing -sqs- patterns as well..

Reply:
Modified the function to test if the existingQueueObj is being passed and in case its passed deadletter queue is not created, also added an integration test case to verify this scenario.